### PR TITLE
feat: better get interaction pactman (CAP-1012)

### PR DESCRIPTION
**Pull request by floredelasteyrie**

### CODE REVIEW

File: src/outcome/devkit/pact.py [hunk 1]
```diff
@@ -40,5 +45,51 @@ def and_given(self, provider_state: str, **params: str):
...
+            interaction_method = interaction['request']['method'].lower()
+            interaction_path = interaction['request']['method'].lower()
+            if interaction_method == method.lower() and interaction_path == path.lower():
+                matched_interactions.append(interaction)
+
+        if len(matched_interactions) > 1:
+            raise AssertionError(f'Request {method} at {path} received, several interactions matching')
+        elif not matched_interactions:
+            raise AssertionError(f'Request {method} at {path} received, no interaction matching')
+        return matched_interactions[0]
```
Comment (d-tw) [inline src/outcome/devkit/pact.py:92]: I would have said that this deserves some tests :)

My reasoning: if we break it, we break all of the api service tests...

**Review by d-tw — CHANGES_REQUESTED** [review]

File: test/test_pact.py [hunk 2]
```diff
@@ -1,4 +1,78 @@
...
+    @pytest.mark.usefixtures('other_interaction_on_same_path')
+    def test_get_other_interaction_from_path(self, url: str):
+        response = requests.get(url=url)
+        assert response.status_code == success_status
+        assert response.json() == {'response': 'Also Matched'}
+
+    @pytest.mark.usefixtures('interaction', 'other_interaction_on_same_path')
+    def test_get_interaction_from_path_fail_several(self, url: str):
+        with pytest.raises(AssertionError, match='several interactions matching'):
+            requests.get(url=url)
```
Comment (d-tw) [inline test/test_pact.py:68]: I think we need a test with multiple non-duplicate interactions, where the pacts are set up in multiple different orders - since that's the whole purpose of the improved matcher

**Review by d-tw — CHANGES_REQUESTED** [review]

File: test/test_pact.py (hunk 2 above)
Comment (floredelasteyrie) [reply test/test_pact.py:68]: True!

**Review by d-tw — APPROVED** [review]
